### PR TITLE
KBD8X QMK Support

### PR DESCRIPTION
**Pull request by mechmerlin**
This PR covers inclusion of the KBD 8X keyboard from KBDFans. It was ported from info available on qmkeyboard.cn

QMK Configurator support for LAYOUT_all and your standard TKL ANSI. None of that ISO stuff!! =P

### CODE REVIEW

File: keyboards/kbd8x/readme.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# KBD 8X
+
+![kbd8x](imgur.com image replace me!)
+
+TKL Keyboard from KBDFans
+
+Keyboard Maintainer: [MechMerlins](https://github.com/mechmerlin)  
+Hardware Supported: kbd8x
```
Comment (drashna) [inline keyboards/kbd8x/readme.md]: Needs an extra line feed/whatever here, as it's staying on one line.

File: keyboards/kbd8x/rules.mk [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#   Teensy halfKay   512
+#   Teensy++ halfKay 1024
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/kbd8x/rules.mk]: BOOTLOADER would be prefered. Caterina, dfu, etc?

File: keyboards/kbd8x/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT_all(
+      KC_ESC,  KC_F1,   KC_F2,   KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,           KC_PSCR, KC_SLCK, KC_PAUS, \
+      KC_GRV,  KC_1,    KC_2,    KC_3,   KC_4,   KC_5,   KC_6,   KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_BSPC, KC_INS,  KC_HOME, KC_PGUP, \
```
[the remark this answers is not in the document]
Comment (mechmerlin) [reply keyboards/kbd8x/keymaps/default/keymap.c]: Don't judge my layouts!!!

File: keyboards/kbd8x/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      KC_TAB,  KC_Q,    KC_W,    KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN, \
+      KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,   KC_G,   KC_H,   KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT, \
+      KC_LSFT, KC_NUBS, KC_Z,    KC_X,   KC_C,   KC_V,   KC_B,   KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                     KC_UP, \
```
[the remark this answers is not in the document]
Comment (mechmerlin) [reply keyboards/kbd8x/keymaps/default/keymap.c]: I guess I could just put a double right shift in there and take out the second layer. I think I got confused as having so many keys kind of defeats the purpose of having layers for my layouts.

File: keyboards/kbd8x/readme.md (hunk 1 above)
Comment (mechmerlin) [reply keyboards/kbd8x/readme.md]: I apparently also spelled my name wrong.

File: keyboards/kbd8x/rules.mk (hunk 2 above)
Comment (mechmerlin) [reply keyboards/kbd8x/rules.mk]: Could you explain this? I don't know what change I need to make. I thought setting it to the right size was all it needed.

Comment (mechmerlin) [reply keyboards/kbd8x/rules.mk]: This is resolved

File: keyboards/kbd8x/keymaps/default/keymap.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (mechmerlin) [reply keyboards/kbd8x/keymaps/default/keymap.c]: Ok I'm saying this is resolved.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**